src/examples/table/TableDraggable.tsx

```diff
@@ -0,0 +1,224 @@
+`;
+
+const StyledCaption = styled(Caption)`
+  margin-bottom: ${props => props.theme.space.sm};
+  line-height: ${props => props.theme.lineHeights.xl};
+  font-size: ${props => props.theme.fontSizes.xl};
+`;
+
+const DraggableContainer = styled.div`
+  :focus {
+    outline: none;
+  }
+`;
+
+interface IDraggableCellProps {
+  isDragOccurring?: boolean;
+}
+
+class DraggableCell extends React.Component<IDraggableCellProps> {
+  ref: HTMLTableCellElement | null = null;
+
+  constructor(args: any) {
+    super(args);
+
+    this.setRef = this.setRef.bind(this);
+  }
+
+  getSnapshotBeforeUpdate(prevProps: IDraggableCellProps) {
+    if (!this.ref) {
+      return null;
+    }
+
+    const isDragStarting = this.props.isDragOccurring && !prevProps.isDragOccurring;
+
+    if (!isDragStarting) {
+      return null;
+    }
+
+    const { width, height } = this.ref.getBoundingClientRect();
+
+    const snapshot = {
+      width,
+      height
+    };
+
+    return snapshot;
+  }
+
+  componentDidUpdate(prevProps: IDraggableCellProps, prevState: any, snapshot: any) {
+    const ref = this.ref;
+
+    if (!ref) {
+      return;
+    }
+
+    if (snapshot) {
+      if (ref.style.width === snapshot.width) {
+        return;
+      }
+
+      ref.style.width = `${snapshot.width}px`;
+      ref.style.height = `${snapshot.height}px`;
+
+      return;
+    }
+
+    if (this.props.isDragOccurring) {
+      return;
+    }
+
+    if (ref.style.width === null) {
+      return;
+    }
+
+    ref.style.removeProperty('height');
+    ref.style.removeProperty('width');
+  }
+
+  setRef(ref: HTMLTableCellElement | null) {
+    this.ref = ref;
+  }
+
+  render() {
+    return <Cell ref={this.setRef}>{this.props.children}</Cell>;
+  }
+}
+
+interface IItem {
+  id: string;
+  content: string;
+}
+
+const generateItems = (count: number) =>
+  Array.from({ length: count }, (v, k) => k).map<IItem>(k => ({
+    id: `item-${k}`,
+    content: `item ${k}`
+  }));
+
+const reorderItems = (list: IItem[], startIndex: number, endIndex: number) => {
+  const result = Array.from(list);
+  const [removed] = result.splice(startIndex, 1);
+
+  result.splice(endIndex, 0, removed);
+
+  return result;
+};
+
+const Example = () => {
+  const [items, setItems] = useState(generateItems(10));
+  const [recentDragId, setRecentDragId] = useState<string>();
+
+  useEffect(() => {
+    if (recentDragId) {
+      const draggable = document.getElementById(recentDragId);
+
+      draggable && draggable.focus();
+    }
+  }, [recentDragId]);
+
+  const onDragEnd = useCallback(
+    (result: DropResult) => {
+      if (!result.destination) {
+        return;
+      }
+
+      const newItems = reorderItems(items, result.source.index, result.destination.index);
+
+      setItems(newItems);
+      setRecentDragId(result.draggableId);
+    },
+    [items]
+  );
+
+  return (
+    <DragDropContext onDragEnd={onDragEnd}>
+      <Table>
+        <StyledCaption>Your Unsolved Tickets</StyledCaption>
```

Comment:
We can drop the `Caption` here. There's already an example to cover that.

Reply:
Caption removed